src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb

```diff
@@ -550,28 +561,63 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
             End If
 
             Dim Here = 1
-            While CanGet(Here)
-                ch = Peek(Here)
-                If IsWhitespace(ch) Then
-                    Here += 1
+            Dim atNewLine As Boolean
+            ' Line continuation is valid at the end of the line, or at the end of the file, or followed by a trailing comment.
+            ' Eg.  LineContinuation ( EndOfLine | EndOfFile | LineContinuationComment)
+            If CanGet(Here) AndAlso (IsWhitespace(Peek(Here)) OrElse IsSingleQuote(Peek(Here))) Then
+                While CanGet(Here)
+                    ch = Peek(Here)
+                    If IsWhitespace(ch) Then
+                        Here += 1
+                    Else
+                        Exit While
+                    End If
+                End While
+                If CanGet(Here) AndAlso IsSingleQuote(Peek(Here)) Then
+                    ' If you get here you have a Line Continuation with have 0 or more spaces followed by a  comment
+                    tList.Add(MakeLineContinuationTrivia(GetText(1)))
+                    If Here > 1 Then
+                        tList.Add(MakeWhiteSpaceTrivia(GetText(Here - 1)))
+                    End If
+                    Dim comment As SyntaxTrivia = ScanComment()
+                    If Not CheckFeatureAvailability(Feature.CommentsAfterLineContinuation) Then
+                        comment = DirectCast(comment.SetDiagnostics({ErrorFactory.ErrorInfo(ERRID.ERR_CommentsAfterLineContinuationNotAvailable1,
+                                                                                                    New VisualBasicRequiredLanguageVersion(Feature.CommentsAfterLineContinuation.GetLanguageVersion()))}), SyntaxTrivia)
+                    End If
+                    tList.Add(comment)
+                    ch = Peek()
```

Comment:
>ch = Peek() [](start = 20, length = 11)

Is it always safe to ```Peek()``` here? What if we are at the EOF?

Reply:
> Yes Peek() returns vbNullChar at EOF

I do not see this behavior documented. Why do you think so?

---
In reply to: [213527098](https://github.com/dotnet/roslyn/pull/28327#discussion_r213527098) [](ancestors = 213527098)

Reply:
I wrote a new test (Scanner_LineContWhenExpectingNewStatement) and there are 2 existing tests(ParseWarningDirective_LineContinuation4V16, ParseWarningDirective_LineContinuation2) for it and then stepped through the code. peek(number) throws an exception but not peek at least in this case. I could put a CanGet around it in case the behavior changes since it is not documented. I think this works because C# puts a 0 at the end of a string buffer and the underlying allocations are all in C#. There are also places in Scanner where the following code is used but always where Here is know to be >0.

```
 ch = If(CanGet(Here), Peek(Here), ChrW(0))
```